src/formulaCompiler.js

```diff
@@ -0,0 +1,638 @@
+const libUtility = require('./utility.js')
+
+export var FormulaCompiler = {
```

Comment:
What is this? Is there any possible way we can delete it? Looking over this file makes me think this library is trying to do WAY too much.

Reply:
This is to allow custom formula based properties to be compiled and ran on the Shader. I think it is a pretty powerful feature to have, and I recommend keeping it until we are sure we don't want it anymore. For now, I have cleaned it up together with the rest of the code.